src/topology/algebra/valuation.lean

```diff
@@ -13,19 +13,18 @@ import ring_theory.valuation.basic
 
 In this file, we define the non archimedean topology induced by a valuation on a ring.
 The main definition is a `valued` type class which equips a ring with a valuation taking
-values in a group with zero (living in the same universe). Other instances are then deduced from
-this.
+values in a group with zero. Other instances are then deduced from this.
 -/
 
 open_locale classical topological_space
 open set valuation
 noncomputable theory
 
-universe u
+universes u v
 
 /-- A valued ring is a ring that comes equipped with a distinguished valuation.-/
 class valued (R : Type u) [ring R] :=
-(Γ₀ : Type u)
+(Γ₀ : Type v)
```

Comment:
What's the advantage to this design over the design that uses an `out_param`?
```lean
class valued (R : Type u) [ring R] (Γ₀ : out_param (Type v)) [linear_ordered_comm_group_with_zero Γ₀] :=
(v : valuation R Γ₀)
```

Reply:
I don't know much about `out_param`, could you explain why using it here might be better?

Reply:
@eric-wieser then `valued` would just be `valuation`, right? @PatrickMassot why do we have this variant where Gamma is bundled?

@digama0 the function you're asking for (which replaces a `{u v}` valuation with an "equivalent" `{u u}` one) is [here](https://github.com/leanprover-community/lean-perfectoid-spaces) in the perfectoid project. /blob/95a6520ce578b30a80b4c36e36ab2d559a842690/src/valuation/canonical.lean#L231)

Reply:
The idea is that if your typeclass is as above, you can use
```lean
example {R Γ₀} [linear_ordered_comm_group_with_zero Γ₀] [valued R Γ₀] : sorry := sorry
```
which reads "assume there is a canonical valuation on type `R` taking values in `Γ₀`". `out_param` tells lean that when applying your lemma it will pick `Γ₀` based on the `valued` instance it found (which is what you want if `Γ₀` is truly canonical).

By contrast the current spelling,
```lean
example [valued R] : sorry := sorry
```
reads "assume there is a canonical valuation on type `R` taking values in `_inst_1.Γ₀`". This is much more annoying if you want to talk about `Γ₀` inside the lemma.

Reply:
Eric we have this already; it's a structure and it's called `valuation` in `ring_theory/valuation/basic`. The point of `valued` is that we use it precisely when we don't want to talk about `Γ₀`, because we only care about things like the topology on `R` generated by the valuation. The philosophy is that we talk about `valuation R Γ₀` when we're talking about valuations and we care about the map `R -> Γ₀`, but for a fixed `R` there's an equivalence relation on the disjoint union of all these valuation types called "equivalence of valuations", basically generated by maps between different choices of `Γ₀`, and we use `valued` when we only care about the equivalence class. 

If this doesn't deal with your worries, can you explain how we can go forward from here?

Reply:
I don't know enough here to make the call, I was just responding to the request to explain `out_param`.

Do we have the results about that equivalence relation, either in master or on a branch? If all you care about is the equivalence relation, should `valued` be a quotient type of some kind? Comparing the options:
1. Hold `Γ₀` in a field with universe `v`
   * Equality of `valued` instances with `valued.Γ₀`s in different universes is unstateable 
   * Equality of `valued` instances with a different choice of `valued.Γ₀` is usually unprovable
2. Hold `Γ₀` in an out_param with universe `v`.
   * Equality of `valued` instances is not stateable, irrespective of universes
3. Hold `Γ₀` in a quotient with universe `v`
   * Equality of `valued` instances with `Γ₀`s in different universes is unstateable
   * Equality of `valued` instances with different types in the same universe is provable
4. Hold `Γ₀` in a quotient with universe `u` (matching `R`)
   * Equality is always provable

I'm happy to defer to whatever @digama0 or another maintainer agrees makes the most sensible

Reply:
> Do we have the results about that equivalence relation, either in master or on a branch?

@eric I think @kbuzzard is referring to [results in the perfectoid space project](https://github.com/leanprover-community/lean-perfectoid-spaces/blob/95a6520ce578b30a80b4c36e36ab2d559a842690/src/valuation/canonical.lean#L13) which we do not yet have in Mathlib

Reply:
@mariainesdff Thank you for this PR; sorry we've been so slow resolving it.

We are happy to make this change to allow allow `Γ₀` to live in a different universe but while you're here could you also:

- Change to the `out_param` design suggested by Eric above.
- Add a comment in the doc string explaining that the purpose of `valued` is to cater for the situation that there is a canonical valuation. You might add that `valued` allows such a valuation to be registered as a typeclass and exhibit `valued.topological_space` as an example of an instance making use of this.
- Add a TODO in the doc string mentioning future work to show that there always exists an equivalent valuation taking values in a type belonging to the same universe as the ring.

Reply:
Thank you! I will add these changes.